### PR TITLE
Make `package` filter independent of other parameters in `GET /request?view=collection` API endpoint

### LINKED ISSUE #13075: `/request?view=collection&package=...` search ignores `package` when `project` is omitted
# Issue Description
`/request?view=collection&package=...` search ignores `package` when `project` is omitted.

Expected Result
============
Return all requests that contain the given `package`.

How to Reproduce
===============
`$ osc api '/request?view=collection&project=Base:System&package=bash&states=new,review'`
returns a single record as expected

while

`$ osc api '/request?view=collection&package=bash&states=new,review'`
returns requests for all packages (the `package=bash` filter is ignored)

Please change the package `bash` to something that has an existing request in new or review state at the moment.

Additional information
=================
The query seems to be broken in other ways too:

`osc api '/request?view=collection&states=new,review&user=yecril71pl&roles=creator'`
returns request for the bash package as expected

while when I combine the two queries into one, an empty collection is returned (returning request for the bash package is still expected):
`osc api '/request?view=collection&project=Base:System&package=bash&states=new,review&user=yecril71pl&roles=creator'`

### PR DESCRIPTION
**Pull request by eduardoj**
Fixes #13075.

### CODE REVIEW

File: src/api/app/models/bs_request/find_for/project.rb [hunk 1]
```diff
@@ -2,45 +2,56 @@ class BsRequest
...
+        return unless roles.empty? || roles.include?(source_or_target)
+
+        bs_request_actions_filters = []
+        if project_name.present?
+          bs_request_actions_filters << if subprojects.blank?
+                                          ["bs_request_actions.#{source_or_target}_project = ?", project_name]
+                                        else
+                                          ["bs_request_actions.#{source_or_target}_project like ?", "#{project_name}:%"]
+                                        end
+        end
```
Comment (danidoni) [inline src/api/app/models/bs_request/find_for/project.rb:30]:
```suggestion
      bs_request_actions_filters << if project_name.present? && subprojects.blank?
                                      ["bs_request_actions.#{source_or_target}_project = ?", project_name]
                                    elsif project_name.present?
                                      ["bs_request_actions.#{source_or_target}_project like ?", "#{project_name}:%"]
                                    end
```
I think it's easier to understand this way

Comment (eduardoj) [reply src/api/app/models/bs_request/find_for/project.rb:30]: With your suggestion, if none of the conditions are met, `bs_request_actions_filters` will be `[nil]`. And we don't want that...

Comment (danidoni) [reply src/api/app/models/bs_request/find_for/project.rb:30]: Yeah, it has to be adapted. But the central idea was to flatten the conditions to avid nesting

Comment (eduardoj) [reply src/api/app/models/bs_request/find_for/project.rb:30]: We also check twice unnecessarily for `project_name.present?`. I still don't see the benefit...

Comment (danidoni) [reply src/api/app/models/bs_request/find_for/project.rb:30]: Alright, np

**Review by danidoni — APPROVED** [review]

**Comment by dmach** [comment]
I've just briefly tested this and the change seems to be working fine for me.

**Pull request merged**